pkg/patterns/addon/pkg/status/aggregate.go

```diff
@@ -74,42 +74,32 @@ func (a *aggregator) Reconciled(ctx context.Context, src declarative.Declarative
 
 	unstruct, ok := src.(*unstructured.Unstructured)
 	instance, commonOkay := src.(addonv1alpha1.CommonObject)
-
-	unstructStatus := make(map[string]interface{})
-	var status addonv1alpha1.CommonStatus
-
-	if ok {
-		unstructStatus["Healthy"] = true
-	} else if commonOkay {
-		status = addonv1alpha1.CommonStatus{Healthy: true}
-	} else {
-		return fmt.Errorf("object %T was not an addonv1alpha1.CommonObject", src)
-	}
+	changed := false
 
 	if commonOkay {
+		var status = instance.GetCommonStatus()
 		status.Errors = statusErrors
 		status.Healthy = statusHealthy
 
 		if !reflect.DeepEqual(status, instance.GetCommonStatus()) {
-			instance.SetCommonStatus(status)
+			status.Healthy = statusHealthy
 
 			log.WithValues("name", instance.GetName()).WithValues("status", status).Info("updating status")
-
-			err := a.client.Status().Update(ctx, instance)
-			if err != nil {
-				log.Error(err, "updating status")
-				return err
-			}
+			changed = true
 		}
-	} else {
-		unstructStatus["Healthy"] = true
-		unstructStatus["Errors"] = statusErrors
+	} else if ok {
```

Comment:
Idea for a future PR: I wonder if we can refactor this into our helper methods ... e.g. GetCommonStatus and SetCommonStatus